MobileSDK/sdk/src/main/java/org/lantern/mobilesdk/Lantern.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Created by ox.to.a.cart on 1/28/16.
+ */
+public class Lantern {
+    private static boolean enabled = false;
+
+    private static final Map<String, Tracker> trackersById = new HashMap<>();
+
+    static {
+        // Track extra info about Android for logging to Loggly.
+        Lantern.addLoggingMetadata("androidDevice", android.os.Build.DEVICE);
+        Lantern.addLoggingMetadata("androidModel", android.os.Build.MODEL);
+        Lantern.addLoggingMetadata("androidSdkVersion", "" + android.os.Build.VERSION.SDK_INT + " (" + android.os.Build.VERSION.RELEASE + ")");
+    }
+
+    /**
+     * <p>Starts Lantern at a random port, storing configuration information in the indicated
+     * configDir and waiting up to timeoutMillis for the proxy to come online. If the proxy fails to
+     * come online within the timeout, this throws an exception.</p>
+     *
+     * <p>If a Lantern proxy is already running within this process, that proxy is reused.</p>
+     *
+     * <p>Note - this does not wait for the entire initialization sequence to finish, just for the
+     * proxy to be listening. Once the proxy is listening, one can start to use it, even as it
+     * finishes its initialization sequence. However, initial activity may be slow, so clients with
+     * low read timeouts may time out.</p>
+     *
+     * @param context
+     * @param timeoutMillis how long to wait for proxy to start listening (should be fairly quick)
+     * @param analyticsTrackingId (optional tracking ID for tracking Google analytics)
+     *
+     * @return the {@link go.lantern.Lantern.StartResult} with port information about the started
+     * lantern
+     */
+    public synchronized static go.lantern.Lantern.StartResult enable(Context context, int timeoutMillis, String analyticsTrackingId) {
+        String configDir = new File(context.getFilesDir().getAbsolutePath(), ".lantern").getAbsolutePath();
+        go.lantern.Lantern.StartResult result = enable(configDir, timeoutMillis);
+        if (analyticsTrackingId != null && !enabled) {
+            trackStartSession(context, analyticsTrackingId);
+        }
+        return result;
+    }
+
+    private static go.lantern.Lantern.StartResult enable(String configDir, int timeoutMillis) {
+        try {
+            go.lantern.Lantern.StartResult result = go.lantern.Lantern.Start(configDir, timeoutMillis);
+            String addr = result.getHTTPAddr();
+            String host = addr.split(":")[0];
+            String port = addr.split(":")[1];
+            System.setProperty("http.proxyHost", host);
+            System.setProperty("http.proxyPort", port);
+            System.setProperty("https.proxyHost", host);
+            System.setProperty("https.proxyPort", port);
+            return result;
+        } catch (Exception e) {
+            throw new RuntimeException("Unable to start Lantern: " + e.getMessage(), e);
```

Comment:
I wonder if throwing a ``RuntimeException`` here is too severe. It means Lantern could not be initialized, and there is nothing the client can do to recover. But applications that integrate the SDK may want the option to continue running regardless.